### DISCUSSION TITLE
Setting switch as button?

### DISCUSSION THREAD

**Discussion by rsilves** [arendst/Tasmota#19006] [Support request]
Hi everyone. I'm setting a new Sonoff Mini R2 with Tasmota, and I'm using setoption114 to detach the switch from the relay. I program my Home Assistant to pick on the MQTT message and toggle another device.

I also have a couple of Sonoff M5 and I am using setoption73, so it sends different commands for single, double, triple, quad an penta presses.

Is there any way to configure the switch in the Mini R2 to send these commands when flipping the switch? If I set the switch as a button, I need to flip the switch twice to send the SINGLE command, and if I flip it only once, it sends the HOLD command.

I don't know if it's possible, I just thought I'd ask here for ideas ;)

Thanks!
Rodrigo

**Comment by sfromis** [comment]
Configuring a pin as a Button is meant to be for a momentary push button, which is very different from when a Switch config is meant to be for a locking toggle, staying put in the same position. That's why they work so differently, including a Button config providing features like multi-press and hold. Using SwitchMode, you can significantly change how a Switch input is processed, including deviation from the "locked toggle" concept.

However, if you config as a Button, you should get action with a single push/release, after a short delay while it is waiting to see if it becomes a multi-press.

**Reply by rsilves** [reply]
Thanks @sfromis for your reply.
Yes I understand they work differently. The question here is if there is a switchmode to make the device send "single", "double", "triple" commands similar to what is possible with buttons.
I had played with setoptions before but now I found that setoption 8 sends "Toggle" on simple toggle and "hold" on double toggle. This is useful to me!

Thanks,
Rodrigo

**Reply by sfromis** [reply]
No, there is not. Multi-press is only with Button pins.
